src/main/java/org/example/MessageService.java

```diff
@@ -2,6 +2,7 @@
 
 public interface MessageService {
```

Comment:
It might be Java way recommended by a course to create an interface in any case, but I'm skeptical about interfaces having a single implementations and no hard plans to extend them.

I'm not an idiomatic Java developer, so I'll let Borja decide on that :-)